otelcollector/test/querymetrics/query_metrics_test.go

```diff
@@ -11,11 +14,359 @@ import (
+				warnings, result, err := utils.InstantQuery(PrometheusQueryClient, fmt.Sprintf("up{job=\"%s\"} == 1", job))
+				Expect(err).NotTo(HaveOccurred())
+
+				// Ensure there are no warnings
+				Expect(warnings).To(BeEmpty())
+
+				// Ensure there is at least one result
+				vectorResult, ok := result.(model.Vector)
+				Expect(ok).To(BeTrue(), "result should be of type model.Vector")
+				Expect(vectorResult).NotTo(BeEmpty(), "result should not be empty")
+
+				// Ensure that all results have the 'up' metric with a value of 1
+				for _, sample := range vectorResult {
+					Expect(string(sample.Metric["__name__"])).To(Equal("up"))
+					Expect(sample.Value.String()).To(Equal("1"))
+				}
+			}
+		})
+	})
+
+	Context("When querying metrics", func() {
+		It("should return the expected results for specified metrics in the default job 'cadvisor'", func() {
+			// Define a list of metrics expected to be present in the default job 'cadvisor'
+			expectedMetrics := []string{
+				"container_spec_cpu_period",
+				"container_spec_cpu_quota",
+				"container_cpu_usage_seconds_total",
+				"container_memory_rss",
+				"container_network_receive_bytes_total",
+				"container_network_transmit_bytes_total",
+				"container_network_receive_packets_total",
+				"container_network_transmit_packets_total",
+				"container_network_receive_packets_dropped_total",
+				"container_network_transmit_packets_dropped_total",
+				"container_fs_reads_total",
+				"container_fs_writes_total",
+				"container_fs_reads_bytes_total",
+				"container_fs_writes_bytes_total",
+				"container_memory_working_set_bytes",
+				"container_memory_cache",
+				"container_memory_swap",
+				"container_cpu_cfs_throttled_periods_total",
+				"container_cpu_cfs_periods_total",
+				// "container_memory_usage_bytes",
+				// "kubernetes_build_info",
+			}
+
+			// Run the query for each metric in the default job 'cadvisor'
+			for _, metric := range expectedMetrics {
+				// Construct the query for the current metric
+				query := fmt.Sprintf("%s{job=\"cadvisor\"}", metric)
+
+				// Run the query for the current metric
+				warnings, result, err := utils.InstantQuery(PrometheusQueryClient, query)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Ensure there are no warnings
+				Expect(warnings).To(BeEmpty())
+
+				// Ensure there is at least one result
+				vectorResult, ok := result.(model.Vector)
+				Expect(ok).To(BeTrue(), "result should be of type model.Vector")
+				Expect(vectorResult).NotTo(BeEmpty(), "result should not be empty")
+
+				// Ensure that the current metric is present in the result
+				found := false
+				for _, sample := range vectorResult {
+					if string(sample.Metric["__name__"]) == metric {
+						found = true
+						break
+					}
+				}
+				Expect(found).To(BeTrue(), fmt.Sprintf("Expected metric %q not found", metric))
+			}
+		})
+	})
+
+	It("should return the expected results for specified metrics in the default job 'kubelet'", func() {
```

Comment:
Thanks Kaveesh! My only suggestion would be to turn these tests into a DescribeTable() test with the job name and expected metric list as the parameters and an Entry for each job

Reply:
Thanks Grace, updated PR to reflect this.